### ISSUE TITLE
Crash with Region Unexplored

### ISSUE THREAD

**Issue by BlackAures1** [Cristelknight/Expanded-Ecosphere#66]
https://github.com/UHQ-GAMES-MODS/REGIONS_UNEXPLORED_FORGE/issues/50#issuecomment-1783693963 
https://github.com/UHQ-GAMES-MODS/REGIONS_UNEXPLORED_FORGE/issues/49

forwarding these two to you.

![image](https://github.com/Cristelknight999/Expanded-Ecosphere/assets/43911230/d51200cf-54f3-4e61-8b83-8110c1f8cc83)

**Comment by UHQ-GAMES** [comment]
hey there! thought id drop my findings here! In a crash log from @BlackAures1 and it looks like a FOC between our biomes on WWOO 3.1.9+ in 1.19.2 and it looks to be the same one we had from the 1.20 versions.

`java.lang.IllegalStateException: Feature order cycle found, involved sources: [Reference{ResourceKey[minecraft:worldgen/biome / regions_unexplored:rainforest]=net.minecraft.world.level.biome.Biome@2525cabb}, Reference{ResourceKey[minecraft:worldgen/biome / wythers:jungle_island]=net.minecraft.world.level.biome.Biome@2651cca8}]`

so I looked into the biomes and i found that in the jungle_island json there is
`"minecraft:patch_melon_sparse",`
 `"minecraft:vines"`

while my rainforest and vanilla sparse_jungle have it the other way around.

if you need any other information I'm happy to help out!

**Comment by catter1** [comment]
Oops, I accidentally duplicated this issue in #74 - but I offered a bit of analysis as well, and referenced a different biome. Maybe it will be helpful

**Comment by BlackAures1** [comment]
Any news on this crash?

**Comment by Cristelknight** [comment]
i don't think i'll upload another 1.19.2 version

**Comment by Cristelknight** [comment]
actually i can't even get it to crash. weird

**Comment by catter1** [comment]
At least for me, the issue was in a version I can no longer find posted anywere (3.1.13)

**Comment by catter1** [comment]
Previously, I was able to recreate the crash on 1.19.4 - I don't think it happens in 1.19.2. Although it theoretically could - I'm assuming there's a change in RU that triggers it.

Anyways, I just tried to recreate it, and I got an unrelated crash. Maybe you'd be interested in it.

https://mclo.gs/xcgS1K6

<details open>
<summary>Mods</summary>

- Cloth Config 10.1.105
- FAPI 0.87.2+1.19.4
- Forge Config API Port 6.0.2
- Regions Unexplored 0.4.1+1.19.4
- TerraBlender 2.3.0.22
- WWOO 3.1.13

</details>
